lib/Frontend/CompilerInvocation.cpp

```diff
@@ -57,6 +59,14 @@ void CompilerInvocation::setMainExecutablePath(StringRef Path) {
   llvm::sys::path::append(DiagnosticDocsPath, "share", "doc", "swift",
                           "diagnostics");
   DiagnosticOpts.DiagnosticDocumentationPath = std::string(DiagnosticDocsPath.str());
+  
+  // Compute the path of the YAML diagnostic messages directory files
+  // in the toolchain.
+  llvm::SmallString<128> DiagnosticMessagesDir(Path);
+  llvm::sys::path::remove_filename(DiagnosticMessagesDir);
```

Comment:
Nit: both of `remove_filename` calls here should mention what exactly are they removing form the path for posterity.